Sources/SuperValidator/Validators/Phone Number/PhoneNumber.swift

```diff
@@ -0,0 +1,166 @@
+            switch self {
+            case .invalidCountryCode, .invalidPhoneNumber, .invalidNationalPrefix, .invalidMaxLength, .countryBlacklisted:
+                return nil
+            }
+        }
+    }
+}
+
+extension SuperValidator {
+    internal func phoneValidator(_ phoneNumber: String, options: Option.PhoneNumber = .init()) -> Result<Void, PhoneNumberError> {
+        var countryCode = ""
+        var nationalPrefix = ""
+        switch options.phoneNumberFormatType {
+        case .nanp:
+            guard phoneNumber.matches(Regex.NANPPhoneFormat) else {
+                return .failure(.invalidPhoneNumber)
+            }
+            countryCode = phoneNumber.getCountryCodeNANP()
+        case .epp:
+            guard phoneNumber.matches(Regex.EPPPhoneFormat) else {
+                return .failure(.invalidPhoneNumber)
+            }
+            countryCode = phoneNumber.getCountryCodeEPP()
+        case .international:
+            guard phoneNumber.matches(Regex.internationalPhoneFormat) else {
+                return .failure(.invalidPhoneNumber)
+            }
+            if options.whitelistPhoneNumberCountry.isNotEmpty {
+                options.whitelistPhoneNumberCountry.forEach { phoneNumberCountry in
+                    if phoneNumber.contains(phoneNumberCountry.countryCode) {
+                        countryCode = phoneNumberCountry.countryCode
+                    }
+                    if phoneNumber.contains(phoneNumberCountry.nationalPrefix) {
+                        nationalPrefix = phoneNumberCountry.nationalPrefix
+                    }
+                }
+            }
+            if options.blacklistPhoneNumberCountry.isNotEmpty {
+                options.blacklistPhoneNumberCountry.forEach { phoneNumberCountry in
+                    if phoneNumber.contains(phoneNumberCountry.countryCode) {
+                        countryCode = phoneNumberCountry.countryCode
+                    }
+                }
+            }
+        }
+        // Whitelist Country
+        if options.whitelistPhoneNumberCountry.isNotEmpty {
+            var isFoundPhoneNumberCountryCode = false
+            var isFoundPhoneNumberNationalPrefix = false
+            var isPhoneNumberLengthValid = true
+            options.whitelistPhoneNumberCountry.forEach { phoneNumberCountry in
+                // Phone Number National Prefix
+                if phoneNumberCountry.nationalPrefix == nationalPrefix {
+                    isFoundPhoneNumberNationalPrefix = true
+                }
+                // Phone Number Country Code
+                if phoneNumberCountry.countryCode == countryCode {
+                    isFoundPhoneNumberCountryCode = true
+                }
+                // Phone number max length
+                if phoneNumberCountry.maxLength < phoneNumber.count {
+                    isPhoneNumberLengthValid = false
+                }
+            }
+            if options.isUsingNationalPrefix {
```

Comment:
hmmm I think user can either input 
+62 or 0, if you do this, they can't right?

Reply:
Right now, user can input either 62 or 0 immediately but the need to specify 
I added a property isUsingNationalPrefix, if isUsingNationalPrefix set to true, user can input with ke 0

Reply:
oh so even set to true user still can input both of em, nice.

you have the use case on UT right?